torch/_inductor/select_algorithm.py

```diff
@@ -760,30 +760,14 @@ def __call__(self, name, choices: List[ChoiceCaller], input_nodes, layout):
         def make_benchmark_fn():
             return self.make_benchmark_fn(choices, input_nodes, layout)
 
-        def autotune(choice):
-            benchmark_fn = make_benchmark_fn()
```

Comment:
This part probably needs some explanation (and possibly some input from anyone who's familiar with this code). In the existing implementation, this error handling would seem to apply to both the in-process and sub-process benchmarking paths. However, the sub-process path was never raising any exceptions in the first place -- it's only providing the latency result or "inf" if the sub-process exits unexpectedly. So I moved this error handling to the in-process path to be more honest about what can happen. 

It's not clear to me whether these conditions can happen in the sub-process path or whether they always applied to the in-process path only. (I never saw them in my testing). If these conditions are possible for both in-process and sub-process, and we want to capture them for both paths, then I'd probably want to refactor along these lines:
a) The sub-process logic should capture exceptions and communicate them back to the parent
b) All relevant functions operate on lists of choices and corresponding lists of results

Reply:
cc @shunting314

Reply:
yea, agree that exception handling here is not needed for sub-process case.